frameworks/cassandra/tests/test_tls.py

```diff
@@ -0,0 +1,256 @@
+        'verify-data', commands, dcos_ca_bundle=dcos_ca_bundle)
+
+
+def get_delete_data_job(dcos_ca_bundle: str):
+    query = ' '.join([
+        'TRUNCATE testspace1.testtable1;',
+        'TRUNCATE testspace2.testtable2;',
+        'DROP KEYSPACE testspace1;',
+        'DROP KEYSPACE testspace2;'])
+    return _get_cqlsh_job_over_tls(
+        'delete-data',
+        [_get_cqlsh_for_query(query)],
+        dcos_ca_bundle=dcos_ca_bundle,
+    )
+
+
+def get_verify_deletion_job(dcos_ca_bundle: str):
+    commands = [
+        '{command} | grep "0 rows"'.format(
+            command=_get_cqlsh_for_query('SELECT * FROM system_schema.tables WHERE keyspace_name=\'testspace1\';')),
+        '{command} | grep "0 rows'.format(
+            command=_get_cqlsh_for_query('SELECT * FROM system_schema.tables WHERE keyspace_name=\'testspace2\';')),
+    ]
+    return _get_cqlsh_job_over_tls(
+        'delete-data', commands, dcos_ca_bundle=dcos_ca_bundle)
+
+
+def _get_cqlsh_job_over_tls(
+        name: str,
+        commands: List[str],
+        dcos_ca_bundle: str=None):
+    """
+    Creates a DC/OS job with `cqlsh` utility that will be ready to run commands
+    over a TLS connection.
+    """
+    commands_with_tls_prepare = [
+        'echo -n "$CQLSHRC_FILE" > $MESOS_SANDBOX/cqlshrc',
+        'echo -n "$CA_BUNDLE" > $MESOS_SANDBOX/ca-bundle.crt',
+    ]
+    commands_with_tls_prepare.extend(commands)
+    cmd = ' && '.join(commands_with_tls_prepare)
+
+    job = {
+        'description': 'Integration test job: ' + name,
+        'id': 'test.cassandra.' + name,
+        'run': {
+            'cmd': cmd,
+            'docker': { 'image': 'cassandra:3.0.13' },
+            'env': {
+                'CQLSHRC_FILE': get_cqlsh_tls_rc_config(),
+                'CA_BUNDLE': dcos_ca_bundle,
+            },
+            'cpus': 1,
+            'mem': 512,
+            'user': 'nobody',
+            'restart': { 'policy': 'NEVER' }
+        }
+    }
+    return job
+
+
+def _get_cqlsh_for_query(query: str):
+    """
+    Creates a `cqlsh` command for given query that will be executed over a TLS
+    connection.
+    """
+    return 'cqlsh --cqlshrc="$MESOS_SANDBOX/cqlshrc" --ssl -e "{query}"'.format(
+        query=query)
+
+
+@pytest.mark.tls
+@pytest.mark.sanity
+@pytest.mark.smoke
+def test_tls_connection(cassandra_service_tls, dcos_ca_bundle):
+    """
+    Tests writing, reading and deleting data over a secure TLS connection.
+    """
+    with sdk_jobs.InstallJobContext([
+            get_write_data_job(dcos_ca_bundle),
+            get_verify_data_job(dcos_ca_bundle),
+            get_delete_data_job(dcos_ca_bundle)]):
+
+        sdk_jobs.run_job(get_write_data_job(dcos_ca_bundle))
+        sdk_jobs.run_job(get_verify_data_job(dcos_ca_bundle))
+
+        key_id = os.getenv('AWS_ACCESS_KEY_ID')
+        if not key_id:
+            assert False, 'AWS credentials are required for this test. Disable test with e.g. TEST_TYPES="sanity and not aws"'
```

Comment:
This test is not marked as `aws` and so I don't thing that the hint provided here would work.

Reply:
Added pytest mark.